includes/admin/wc-meta-box-functions.php

```diff
@@ -227,7 +227,10 @@ function woocommerce_wp_select( $field ) {
 		<select <?php echo wc_implode_html_attributes( $field_attributes ); // WPCS: XSS ok. ?>>
 			<?php
 			foreach ( $field['options'] as $key => $value ) {
-				echo '<option value="' . esc_attr( $key ) . '" ' . selected( $field['value'] === $key || ( is_array( $field['value'] ) && in_array( $key, $field['value'], true ) ), true, false ) . '>' . esc_html( $value ) . '</option>';
+				// The $key from $field['options'] could have been converted to integer automatically by PHP,
+				// thus the comparison needs to check both $key and (string) $key in case $field['value'] is an array
+				// with string values.
+				echo '<option value="' . esc_attr( $key ) . '" ' . selected( $field['value'] === $key || ( is_array( $field['value'] ) && ( in_array( $key, $field['value'], true ) || in_array( (string) $key, $field['value'], true ) ) ), true, false ) . '>' . esc_html( $value ) . '</option>';
```

Comment:
Would it be possible to make sure that the comparison also works for regular select fields? See my comment at https://github.com/woocommerce/woocommerce/issues/19619#issuecomment-381055290 for more details.

Reply:
Thx, that's a good point, I missed that since I focused on the multiselect problem.

Reply:
@peterfabian Cast both? As we said, `$key` will be a string when posted. So if we cast the $key and the compared value we only need one set of logic.

Reply:
Well, `meta_value` is a text, so I thought it's safe. But it's true that if value is passed manually instead of fetched from the database, then it would still fail.